### PR TITLE
myeloid rankmodels + loqusdb-fixes for non-SV runs

### PR DESCRIPTION
**Pull request by alkc**
# Description and reviewer info

This update creates temp rank model files for `myeloid_const` to disable loqusdb artefact scoring until artefact db is populated. 

This update also creates a workaround to enable loqusdb loads for profiles/runs where SV calling/annotation is disabled, as is the case for `myeloid_const`

## TODO:

- [x] Stubs for dummy svvcf process
- [x] Document why assay included in dummy svvcf process input

## Type of change

- [x] Minor change

# Checklist

- [x] Self-review of my code
- [x] Update the CHANGELOG
- [ ] Tag the latest commit (vX.Y.Z format)
- [ ] Log samples used for testing in the `Verification_samples_log` Excel sheet

## Documentation
- [ ] At least one other person has reviewed my changes (not required for trivial changes)

## Major / Minor change
- [ ] `onco` run finishes without any new warnings/errors 
- [ ] `wgs` single run finishes without any new warnings/errors
- [ ] `wgs` trio run finishes without any new warnings/errors
- [x] loqusdb load instructions are generated for `myeloid_const` runs
- [x] loqusdb load instructions are generated for `wgs` runs and svvcf path is included as an arg 
- [x] same as above but for an onco run
- [x] check that artefact scoring really is missing from temp rank model files
- [x] At least one other person has reviewed and approved my code
- [ ] I have made corresponding changes to the documentation (software versions, etc.)

# Test/review documentation

## Review performed by

- [ ] Alexander
- [ ] Jakob
- [ ] Paul
- [ ] Ryan
- [ ] Viktor

(Add if missing)

## Testing performed by

- [x] Alexander
- [ ] Jakob
- [ ] Paul
- [ ] Ryan
- [ ] Viktor

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -124,6 +124,12 @@ bam_bqsr_choice.map {
 // 	split_vep_choice;
 // }
 
+Channel
+	.fromPath(params.csv)
+	.splitCsv(header:true)
+	.map{ row-> tuple(row.group, row.assay) }
+        .set{ meta_loqusdb_no_sv_calling }
```
Comment (alkc) [inline main.nf:131]: `assay` here is not required by the process, other than to disable a nextflow warning about a channel emitting only one input to the new process below

File: main.nf [hunk 2]
```diff
@@ -3390,22 +3426,23 @@ process add_to_loqusdb {
 		!params.noupload && !params.reanalyze
 
 	input:
-		set group, type, file(vcf), file(tbi), type, file(ped), file(svvcf) from vcf_loqus.join(ped_loqus).join(loqusdb_sv.mix(loqusdb_sv_panel))
+		set group, type, file(vcf), file(tbi), type, file(ped) from vcf_loqus.join(ped_loqus)
+                set group, file(svvcf) from loqusdb_sv.mix(loqusdb_sv_panel, dummy_svvcf_ch)
```
Comment (alkc) [inline main.nf:3430]:
```suggestion
		set group, type, file(vcf), file(tbi), type, file(ped) from vcf_loqus.join(ped_loqus)
		set group, file(svvcf) from loqusdb_sv.mix(loqusdb_sv_panel, dummy_svvcf_ch)
```

fix ugly whitespace

File: main.nf [hunk 3]
```diff
@@ -3390,22 +3426,23 @@ process add_to_loqusdb {
...
-		else {
-			"""
-			echo "-db $params.loqusdb load -f ${params.accessdir}/ped/${ped} --variant-file ${params.accessdir}/vcf/${vcf} --sv-variants ${params.accessdir}/sv_vcf/merged/${svvcf}" > ${group}.loqus
-			"""
-		}
+                """
+                sv_variants=""
+                nbr_svvcf_records=\$(grep -v '^#' ${svvcf} | wc -l)
+
+                if (( \$nbr_svvcf_records > 0 )); then
```
Comment (ViktorHy) [inline main.nf:3440]: I think this is actually pretty nice way. If mody will end up with no calls for SVs we might end up having to exclude SVs from the import as well

File: configs/nextflow.hopper.config [hunk 4]
```diff
@@ -303,13 +303,13 @@ myeloid {
     params.mei_list = "${params.refpath}/annotation_dbs/MELT/mei_list.txt"
     params.meltheader = "${params.refpath}/annotation_dbs/MELT/melt_vcf_header"
     params.svdb = "${params.refpath}/annotation_dbs/onco-solid/artefact-sv/200513/svdb_200513.db"
-    params.svrank_model_s = "/fs1/resources/scout/rank_models_38/svrank_single_v1.0_panel__VEP111.ini"
-    params.rank_model = "/fs1/resources/scout/rank_models_38/rank_model_cmd_v4__VEP111.ini"
-    params.rank_model_s = "/fs1/resources/scout/rank_models_38/rank_model_cmd_v5.0_single_withoutmodels_onco__VEP111.ini"
-    params.svrank_model = "/fs1/resources/scout/rank_models_38/svrank_model_v1.0__VEP111.ini"
+    params.svrank_model_s = "/fs1/resources/scout/rank_models_38/myeloid_noloqusdb/svrank_single_v1.0_panel__VEP111.ini"
```
Comment (ViktorHy) [inline configs/nextflow.hopper.config:306]: good to keep rank models separated by subdirs, could perhaps make it easier to make sense of all the different files

**Review by ViktorHy — APPROVED** [review]
nothing to add really. Make sure the code works for various profiles

**Comment by alkc** [comment]
Thanks for merging and deploying @Jakob37

**Pull request merged**